packages/dx-react-scheduler-demos/src/demo-sources/scheduler-editing/material-ui/delete-confirmation.jsx

```diff
@@ -66,33 +67,31 @@ export default class Demo extends React.PureComponent {
   }
 
   commitDeletedAppointment() {
-    const { data, deletedAppointmentId } = this.state;
-    const nextData = data.filter(appointment => appointment.id !== deletedAppointmentId);
-    this.setState({ data: nextData, deletedAppointmentId: null });
+    this.setState((state) => {
+      const { data, deletedAppointmentId } = state;
+      const nextData = data.filter(appointment => appointment.id !== deletedAppointmentId);
+      return { data: nextData, deletedAppointmentId: null };
+    });
     this.toggleConfirmationVisibility();
```

Comment:
Here confirmation visibility is toggled before state change is executed while in `commitChanges` down below the same is done together with state change execution. Is it as planned?